ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -787,6 +792,12 @@ public void setSafeBlock(final Hash blockHash) {
     updater.commit();
   }
 
+  public void setEarliest(final Hash blockHash) {
```

Comment:
looks like we are always after the block number when we call getEarliest(). Should we store the block number instead? Then we do not have to make the conversion hash -> block number when we need it.